guice-extension/src/main/java/com/github/kaktushose/jda/commands/guice/Implementation.java

```diff
@@ -1,67 +1,27 @@
 package com.github.kaktushose.jda.commands.guice;
 
-import com.github.kaktushose.jda.commands.annotations.constraints.Constraint;
 import com.github.kaktushose.jda.commands.definitions.description.ClassFinder;
 import com.github.kaktushose.jda.commands.definitions.description.Descriptor;
-import com.github.kaktushose.jda.commands.dispatching.adapter.TypeAdapter;
-import com.github.kaktushose.jda.commands.dispatching.middleware.Middleware;
-import com.github.kaktushose.jda.commands.dispatching.middleware.Priority;
-import com.github.kaktushose.jda.commands.dispatching.validation.Validator;
 import com.github.kaktushose.jda.commands.embeds.error.ErrorMessageFactory;
 import com.github.kaktushose.jda.commands.permissions.PermissionsProvider;
 import com.github.kaktushose.jda.commands.scope.GuildScopeProvider;
 import jakarta.inject.Scope;
 
-import java.lang.annotation.*;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
 
 /// Indicates that the annotated class is a custom implementation that should replace the default implementation.
 ///
 /// A class annotated with [Implementation] will be automatically searched for with help of the [ClassFinder]s
 /// and instantiated by guice. Following types are candidates for automatic registration.
 ///
-/// - [Middleware]
-/// - [Validator]
-/// - [TypeAdapter]
 /// - [PermissionsProvider]
 /// - [GuildScopeProvider]
 /// - [ErrorMessageFactory]
 /// - [Descriptor]
-///
-/// ### Example
-/// ```java
-/// @Implementation(priority = Priority.NORMAL)
-/// public class CustomMiddleware implements Middleware {
-///     private static final Logger log = LoggerFactory.getLogger(FirstMiddleware.class);
-///
-///     @Override
-///     public void accept(InvocationContext<?> context) {
-///         log.info("run custom middleware");
-///     }
-/// }
-/// ```
```

Comment:
We should note, that Type Adapters, Validators and Middlewares can be registered via their specific annotations.